eox_tenant/signals.py

```diff
@@ -217,3 +222,114 @@ def clear_tenant(sender, request, **kwargs):  # pylint: disable=unused-argument
     Signal: django.core.signals.got_request_exception
     """
     pass
+
+
+def tenant_context_addition(sender, body, headers, *args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Receiver used to add context to the async process about the tenant associated to the task that need to be done.
+
+    Dispatched before a celery task is published. Note that this is executed in the process sending the task.
+    See:
+        https://celery.readthedocs.io/en/latest/userguide/signals.html#before-task-publish
+    """
+    get_host_func = AsyncTaskHandler().get_host_from_task(sender)
+    headers['eox_tenant_sender'] = get_host_func(body)
+
+
+def start_async_tenant(sender, *args, **kwargs):  # pylint: disable=unused-argument
+    """
+    Receiver that runs on the async process to update the settings accordingly to the tenant.
+    Dispatched before a task is executed.
+    See:
+       https://celery.readthedocs.io/en/latest/userguide/signals.html#task-prerun
+    """
+    headers = sender.request.get('headers') or {}
+    http_host = headers.get('eox_tenant_sender')
+    if not http_host:
+        LOG.warning("Could not find the host information for eox_tenant.signals. ")
+        siteid = base_settings.SITE_ID
+        site = Site.objects.get(id=siteid)
+        http_host = site.name
+
+    try:
+        domain = http_host.split(':')[0]
```

Comment:
you need this domain to execute the method `_update_settings` but `_update_settings`, after a lot of method calls, executes this line with the domain parameter https://github.com/eduNEXT/eox-tenant/blob/master/eox_tenant/models.py#L149, so why don't remove this or add this to the `get_host` methods that returns the domain